### PR TITLE
CXP-765 remove App connection from connections settings UI

### PR DESCRIPTION
**Pull request by RiffVentura**
Prevented display of non-default connections in the connection settings UI
Added a type property to connections that defaults to 'default'

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 1]
```diff
@@ -142,6 +142,9 @@ akeneo_connectivity.connection:
                 invalid: Connection flow type can only be 'data_source', 'data_destination' or 'other'.
             image:
                 must_exist: The image was not found. Please try again.
+            type:
+                required: Connection type is required.
+                too_long: Connection type is too long. It should have 100 characters or less.
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml:147]:
```suggestion
                too_long: Connection type is too long. It should have 30 characters or less.
```

**Review by tseho — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Domain\Settings\Model\ValueObject;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ConnectionType
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php:11]: final

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const DEFAULT_TYPE = 'default';
+
+    private string $type;
+
+    public function __construct(?string $type = self::DEFAULT_TYPE)
+    {
+        $type = $type ?? self::DEFAULT_TYPE;
+
+        if (empty($type)) {
+            throw new \InvalidArgumentException('akeneo_connectivity.connection.connection.constraint.type.required');
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php:22]: As it's not possible, I would throw a LogicException

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        if (mb_strlen($type) > 30) {
+            throw new \InvalidArgumentException('akeneo_connectivity.connection.connection.constraint.type.too_long');
+        }
+
+        $this->type = $type;
+    }
+
+    public function __toString()
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php:32]:
```suggestion
    public function __toString(): string
```
I think it is still required even if you use the magical function.

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    public function __construct(?string $type = self::DEFAULT_TYPE)
+    {
+        $type = $type ?? self::DEFAULT_TYPE;
+
+        if (empty($type)) {
+            throw new \InvalidArgumentException('akeneo_connectivity.connection.connection.constraint.type.required');
+        }
+
+        if (mb_strlen($type) > 30) {
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php:25]: Why?

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Repository/DbalConnectionRepositoryIntegration.php [hunk 6]
```diff
@@ -73,6 +75,37 @@ public function test_it_updates_a_connection_from_its_code()
...
+        $result = $this->selectConnectionFromDb('new_connection');
+
+        Assert::assertSame('new_connection', $result['code']);
+        Assert::assertSame('new_connection_label', $result['label']);
+        Assert::assertSame(FlowType::OTHER, $result['flow_type']);
+        Assert::assertNotNull($result['client_id']);
+        Assert::assertNotNull($result['user_id']);
+        Assert::assertSame(true , (bool) $result['auditable']);
+        Assert::assertSame('connection_type' , $result['type']);
+
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Repository/DbalConnectionRepositoryIntegration.php:106]: rm blank line

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Repository/DbalConnectionRepositoryIntegration.php [hunk 7]
```diff
@@ -73,6 +75,37 @@ public function test_it_updates_a_connection_from_its_code()
...
+        $this->repository->create($connection);
+
+        $result = $this->selectConnectionFromDb('new_connection');
+
+        Assert::assertSame('new_connection', $result['code']);
+        Assert::assertSame('new_connection_label', $result['label']);
+        Assert::assertSame(FlowType::OTHER, $result['flow_type']);
+        Assert::assertNotNull($result['client_id']);
+        Assert::assertNotNull($result['user_id']);
+        Assert::assertSame(true , (bool) $result['auditable']);
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Repository/DbalConnectionRepositoryIntegration.php:104]: FYI `assertTrue` exists

**Review by willy-ahva — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php (hunk 3 above)
Comment (tseho) [reply src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionType.php:22]: I'm going to leave it like this, he wanted to be consistent with the other ValueObject in the same directory.

**Review by Airmanbzh — APPROVED** [review]

**Pull request merged**